bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/util/ActionInputsHelper.java

```diff
@@ -0,0 +1,327 @@
+        String defaultValue = null;
+        Unit<?> unit = null;
+        boolean required = false;
+        String context = null;
+        Matcher matcher = QUANTITY_TYPE_PATTERN.matcher(input.getType());
+        if (matcher.matches()) {
+            parameterType = ConfigDescriptionParameter.Type.DECIMAL;
+            try {
+                unit = getDefaultUnit(matcher.group("dimension"));
+            } catch (IllegalArgumentException e) {
+                supported = false;
+            }
+        } else {
+            switch (input.getType()) {
+                case "boolean":
+                    defaultValue = "false";
+                    required = true;
+                case "java.lang.Boolean":
+                    parameterType = ConfigDescriptionParameter.Type.BOOLEAN;
+                    break;
+                case "byte":
+                case "short":
+                case "int":
+                case "long":
+                    defaultValue = "0";
+                    required = true;
+                case "java.lang.Byte":
+                case "java.lang.Short":
+                case "java.lang.Integer":
+                case "java.lang.Long":
+                    parameterType = ConfigDescriptionParameter.Type.INTEGER;
+                    break;
+                case "float":
+                case "double":
+                    defaultValue = "0";
+                    required = true;
+                case "java.lang.Float":
+                case "java.lang.Double":
+                case "java.lang.Number":
+                case "org.openhab.core.library.types.DecimalType":
+                    parameterType = ConfigDescriptionParameter.Type.DECIMAL;
+                    break;
+                case "java.lang.String":
+                    break;
+                case "java.time.LocalDate":
+                    context = "date";
+                    break;
+                case "java.time.LocalTime":
+                    context = "time";
+                    break;
+                case "java.time.LocalDateTime":
+                case "java.util.Date":
+                    context = "datetime";
+                    break;
+                case "java.time.ZonedDateTime":
+                case "java.time.Instant":
+                case "java.time.Duration":
+                    // There is no available configuration parameter context for these types.
+                    // A text parameter is used. The expected value must respect a particular format specific
+                    // to each of these types.
+                    break;
+                default:
+                    supported = false;
+                    break;
+            }
+        }
+        if (!supported) {
+            throw new IllegalArgumentException("Input parameter '" + input.getName() + "' with type " + input.getType()
+                    + "cannot be converted into a config description parameter!");
+        }
```

Comment:
```suggestion
                    throw new IllegalArgumentException("Input parameter '" + input.getName() + "' with type " + input.getType()
                    + "cannot be converted into a config description parameter!");
            }
        }
```

You can also remove the `supported` variable above.

Reply:
Done